blueprints/ember-cli-eslint/index.js

```diff
@@ -1,9 +1,187 @@
+/*eslint-env node*/
+
+var unlink = require('fs').unlink;
+var resolve = require('path').resolve;
+
+var RSVP = require('rsvp');
+var walkSync = require('walk-sync');
+
+/**
+ * For each item in the array, call the callback, passing the item as the argument.
+ * However, only call the next callback after the promise returned by the previous
+ * one has resolved.
+ *
+ * @param {*[]} items the elements to iterate over
+ * @param {Function} cb called for each item, with the item as the parameter. Should return a promise
+ * @return {Promise}
+ */
+function synchronize(items, cb) {
+  return items.reduce(function(promise, item) {
+    return promise.then(function() {
+      return cb.call(this, item);
+    });
+  }, RSVP.resolve());
+}
+
 module.exports = {
   name: 'ember-cli-eslint',
 
   normalizeEntityName: function() {
     // this prevents an error when the entityName is
     // not specified (since that doesn't actually matter
     // to us
+  },
+
+  afterInstall: function() {
+    var removeJSHintConfig = this._removeJSHintConfig.bind(this);
+
+    if (!this.removePackageFromProject) {
+      return;
+    }
+
+    return this.removePackageFromProject('ember-cli-jshint').then(function() {
+      return removeJSHintConfig();
+    });
```

Comment:
I think it might be more readable if you change this to:

```js
return this.removePackageFromProject('ember-cli-jshint').then(function() {
  return this._removeJSHintConfig();
}.bind(this));
```